### PR TITLE
fix: use explicit buffer id to ensure the behavior of 'winleave' auto…

### LINKED ISSUE #36: Invalid window id error when press q to quit
when press q to quit or press enter to open a file in yazi.nvim ui, invalid window id error pops up. here is the detail:

trace
```
	/home/ak2kay/.local/share/nvim/lazy/yazi.nvim/lua/yazi.lua:48: in function </home/ak2kay/.local/share/nvim/lazy/yazi.nvim/lua/yazi.lua:41> function: builtin#18 ...2kay/.local/share/nvim/lazy/yazi.nvim/lua/yazi/utils.lua:194: Invalid window id: 1372
stack traceback:
	[C]: in function 'nvim_win_close'
	...2kay/.local/share/nvim/lazy/yazi.nvim/lua/yazi/utils.lua:194: in function 'on_yazi_exited'
	/home/ak2kay/.local/share/nvim/lazy/yazi.nvim/lua/yazi.lua:48: in function </home/ak2kay/.local/share/nvim/lazy/yazi.nvim/lua/yazi.lua:41>
```

versions:
```
❯ yazi --version
Yazi 0.2.4 (Homebrew 2024-04-17)
❯ nvim --version
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1710088188

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/home/linuxbrew/.linuxbrew/Cellar/neovim/0.9.5/share/nvim"

Run :checkhealth for more info
```

config:
```lua
{
    "mikavilpas/yazi.nvim",
    dependencies = {
      "nvim-lua/plenary.nvim",
    },
    event = "VeryLazy",
    keys = {
      {
        -- 👇 choose your own keymapping
        "<leader>fy",
        function()
          require("yazi").yazi()
        end,

        { desc = "Open the file manager" },
      },
    },
    opts = {
      open_for_directories = false,
    },
  }
```

### PR DESCRIPTION
**Pull request by ak2kay**
mainly fixes #36 

in my case, there are two problems i met: 
* the first one is when i call open method the first time, after i choosen the file, the border window won't close, a screenshot is attached below; 
* and the second one is the the same as the issue i created, the border window is closed unexpectedly before we call nvim_win_close. i am not sure why this is happening, but add a check to avoid it.

图片

### CODE REVIEW

**Comment by mikavilpas** [comment]
Thanks, I think this is a good idea. I think it might need a bit more before it can be merged: I noticed that if I click outside of the yazi modal, the window stays up but its contents (yazi in the terminal) are now killed.

I don't think many people will have this issue, but it would be easier for me to solve that at the same time as this. I will do some more testing hopefully today.

**Comment by mikavilpas** [comment]
Note to myself: also there seem to be some issues when opening a file in a new split with this change. It can probably be solved though.

**Comment by mikavilpas** [comment]
Ok I refactored the window implementation a bit. Now it should try to safely close when focus is lost - previously I guess sometimes one of the buffers was already closed due to your other plugin having its own autocommands (not really sure)

Can you try this implementation out?

**Comment by mikavilpas** [comment]
Ah never mind, it doesn't work yet. Let me try again.

**Comment by mikavilpas** [comment]
Oh it actually does work, the reproduction file just ran the unfixed code. Here's an updated one that targets this branch (in your fork). You might be able to automatically update it with `nvim -u repro.lua -c "lua require('lazy').update()"`. If that doesn't work then you have to call `:Lazy` and use the ui to `U`pdate, etc.

<details><summary>Details</summary>
<p>

```lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify('./.repro', ':p')

-- set stdpaths to use .repro
for _, name in ipairs({ 'config', 'data', 'state', 'cache' }) do
  vim.env[('XDG_%s_HOME'):format(name:upper())] = root .. '/' .. name
end

-- bootstrap lazy
local lazypath = root .. '/plugins/lazy.nvim'
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    'git',
    'clone',
    '--filter=blob:none',
    'https://github.com/folke/lazy.nvim.git',
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)
vim.g.mapleader = ' '

-- install plugins
local plugins = {
  'folke/tokyonight.nvim',
  {
    'ak2kay/yazi.nvim',
    branch = 'fix-window',
    dependencies = {
      'nvim-lua/plenary.nvim',
    },
    event = 'VeryLazy',
    keys = {
      {
        -- 👇 choose your own keymapping
        '<leader>fy',
        function()
          require('yazi').yazi()
        end,

        { desc = 'Open the file manager' },
      },
    },

    opts = {
      open_for_directories = false,
    },
  },
  { 'zbirenbaum/nvterm', opts = {} },
}
require('lazy').setup(plugins, {
  root = root .. '/plugins',
})

vim.cmd.colorscheme('tokyonight')
-- add anything else here
```

</p>
</details>

**Comment by ak2kay** [comment]
That's very kind and nice of you! 

I can confirm my config is working as expected with your refactoring.
